### PR TITLE
Adding support for ssh certificates.

### PR DESCRIPTION
**Pull request by vmasutin**
Tested with github, please see
https://github.blog/2019-08-14-ssh-certificate-authentication-for-github-enterprise-cloud/
for more details.

Signed-off-by: Valery Masiutsin <val.masutin@gmail.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by radu-matei** [comment]
ref #407 

/brig run

**Comment by radu-matei** [comment]
Hi!
Thanks a lot for picking this up!
I know this is still a WIP, just a heads up to go over the Checks tab as there are a few test failures across the controller and worker.
Thanks!

**Comment by vmasutin** [comment]
tests passed for me locally but will probably fail in CI (because of pending brigadier change [brigadier PR](https://github.com/brigadecore/brigadier/pull/33)).
Its a bit of a chicken & egg problem. Let me know your prefered way to resolve this.

File: brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts [hunk 1]
```diff
@@ -129,6 +129,8 @@ export interface Repository {
      * sshKey the SSH key to use for ssh:// or git+ssh:// protocols
      */
     sshKey?: string;
+
+    sshCert?: string;
```
Comment (vdice) [inline brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts]: This file was amended to get tests passing, right?  Let's remember to remove this change (perhaps after bumping the brigadier version if https://github.com/brigadecore/brigadier/pull/33 is merged first).

File: pkg/brigade/project_test.go [hunk 2]
```diff
@@ -42,6 +43,9 @@ func TestProjectSecrets(t *testing.T) {
 	if got.Repo.SSHKey != "" {
 		t.Error("Project.Repo.SSHKey should not be exported")
 	}
+	if got.Repo.SSHCert != "" {
+		t.Error("Project.Repo.SSHKey should not be exported")
```
Comment (vdice) [inline pkg/brigade/project_test.go]: copy/paste typo in err message here... `s/SSHKey/SSHCert`

File: brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts (hunk 1 above)
Comment (vmasutin) [reply brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts]: correct, or we can add temporary branch to brigadier and use it in  yarn, but thats probably an overkill...

Comment (vdice) [reply brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts]: Ok, now that https://github.com/brigadecore/brigadier/pull/33 is merged, let me get an npm release out there and then we can update this branch and we should be looking good... will report back once release is cut.

Comment (vmasutin) [reply brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts]: cool

Comment (vdice) [reply brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts]: Ok @vmasutin - https://www.npmjs.com/package/@brigadecore/brigadier/v/0.9.0 is out.  Please update the dep in this branch and rm the manual override.  Let's run CI once that is done.  Thanks!

Comment (vmasutin) [reply brigade-worker/node_modules/@brigadecore/brigadier/out/events.d.ts]: I have pushed the change

**Comment by vdice** [comment]
@vmasutin Thanks for your patience on the brigadier release, etc.  A few items left, when convenient:

* Can you run a `make yarn-install` and then check in the `yarn.lock` and `node_modules` changes?  That should get the js checks passing...

* Also, can we add a mention around the new `BRIGADE_REPO_SSH_CERT` env var to the corresponding doc in https://github.com/brigadecore/brigade/blob/master/docs/content/topics/workers.md#project-level-environment-variables ?

**Pull request merged**